gen_breaktest.go

```diff
@@ -0,0 +1,215 @@
+// +build generate
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"go/format"
+	"io"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"strings"
+)
+
+//go:generate go run gen_breaktest.go
+
+// See https://www.unicode.org/license.html for the Unicode license agreement.
+
+// We want to test against a specific version rather than the latest, which
+// can be found at:
+// https://www.unicode.org/Public/UCD/latest/ucd/auxiliary/GraphemeBreakTest.txt
+// When/if the package is upgraded to a new version, change these to generate new tests.
+const (
+	url      = `https://www.unicode.org/Public/13.0.0/ucd/auxiliary/GraphemeBreakTest.txt`
+	filename = `GraphemeBreakTest-13.0.0.txt`
+)
+
+func main() {
```

Comment:
Can we have a comment here what this program does?

Reply:
Absolutely, when I just now looked at the Go sources I listed above there doesn't seem to be any standard they follow for this. Based on the ones saw that I liked, how about putting this at the top between the `// +build …` and `package main` lines:

    // This program generates grapheme_break_test.go from the Unicode Character
    // Database auxiliary data files at https://www.unicode.org/Public/
    // Either directly via HTTP by URL or from a local copy of the file.
    //
    //go:generate go run gen_breaktest.go

Reply:
Looks good to me now!